Modules/_multiprocessing/posixshmem.c

```diff
@@ -0,0 +1,724 @@
+    0,                                  // tp_setattro
+    0,                                  // tp_as_buffer
+    Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
+                                        // tp_flags
+    "POSIX shared memory object",       // tp_doc
+    0,                                  // tp_traverse
+    0,                                  // tp_clear
+    0,                                  // tp_richcompare
+    0,                                  // tp_weaklistoffset
+    0,                                  // tp_iter
+    0,                                  // tp_iternext
+    SharedMemory_methods,               // tp_methods
+    SharedMemory_members,               // tp_members
+    SharedMemory_getseters,             // tp_getset
+    0,                                  // tp_base
+    0,                                  // tp_dict
+    0,                                  // tp_descr_get
+    0,                                  // tp_descr_set
+    0,                                  // tp_dictoffset
+    (initproc) SharedMemory_init,       // tp_init
+    0,                                  // tp_alloc
+    (newfunc) SharedMemory_new,         // tp_new
+    0,                                  // tp_free
+    0,                                  // tp_is_gc
+    0                                   // tp_bases
+};
+
+
+/*
+ *
+ * Module-level functions & meta stuff
+ *
+ */
+
+static PyObject *
+posixshmem_unlink_shared_memory(PyObject *self, PyObject *args) {
+    const char *name;
+
+    if (!PyArg_ParseTuple(args, "s", &name))
+        return NULL;
+    else
+        return my_shm_unlink(name);
+}
+
+
+static PyMethodDef module_methods[ ] = {
+    {   "unlink_shared_memory",
+        (PyCFunction)posixshmem_unlink_shared_memory,
+        METH_VARARGS,
+        "Unlink shared memory"
+    },
+    {NULL} /* Sentinel */
+};
+
+
+static struct PyModuleDef this_module = {
+    PyModuleDef_HEAD_INIT,  // m_base
+    "_posixshmem",          // m_name
+    "POSIX shared memory module",     // m_doc
+    -1,                     // m_size (space allocated for module globals)
+    module_methods,         // m_methods
+    NULL,                   // m_reload
+    NULL,                   // m_traverse
+    NULL,                   // m_clear
+    NULL                    // m_free
+};
+
+/* Module init function */
+PyMODINIT_FUNC
+PyInit__posixshmem(void) {
+    PyObject *module;
+    PyObject *module_dict;
+
+    // I call this in case I'm asked to create any random names.
+    srand((unsigned int)time(NULL));
+
+    module = PyModule_Create(&this_module);
+
+    if (!module)
+        goto error_return;
+
+    if (PyType_Ready(&SharedMemoryType) < 0)
+        goto error_return;
+
+    Py_INCREF(&SharedMemoryType);
+    PyModule_AddObject(module, "_PosixSharedMemory", (PyObject *)&SharedMemoryType);
+
+
+    PyModule_AddStringConstant(module, "__copyright__", "Copyright 2012 Philip Semanchuk, 2018-2019 Davin Potts");
```

Comment:
The only other files with a __copyright__ attribute are parser, optparse and platform. I'd prefer to avoid adding new copyright attributes to stdlib modules.

Reply:
See comment on issue.

Reply:
Removed in 7c6501781744c4960eaa4991aba2cd73f3c7dae0 which should appear in GH-11816 shortly.

I have pinged Van again asking for verification that this is indeed appropriate but I have made the change in anticipation that this is all good.